Makefile

```diff
@@ -311,6 +324,19 @@ check-vera:
 check-cppcheck:
 	$(Q) $(call SHLOG,./tools/check-cppcheck.sh,$(OUT_DIR)/cppcheck.log,Cppcheck)
 
+.PHONY: test-buildoptions
+test-buildoptions:
+	$(Q) for target in $(JERRY_BUILD_OPTIONS_TEST_TARGETS); do \
+           if [[ $$target =~ "mcu" ]]; then \
+             current_options=$$BUILD_OPTIONS_TEST_MCU; \
+           else \
+             current_options=$$BUILD_OPTIONS_TEST_NATIVE; \
+           fi; \
+           for option in $$current_options; do \
+             $(MAKE) $$option=ON $$target || exit 1; \
+           done \
+         done
```

Comment:
What still bothers me is that these for loops are still sequential. Cannot we declare a make target for each of the `$(MAKE) $$option=ON $$target` commands and have all those targets be a dependecy of test-buildoptions? Then make would take care of parallelization, if it's available on the system. (Ex: test-js.*.* targets)